### PR TITLE
SorrTask1020_compute_hit_rate

### PR DESCRIPTION
**Pull request by tkpratardan**
- Compute hit rates and assess them in a 95% confidence interval.
- Performed a paired-test using a sample size of 7000 and level of significance of of p < 0.05 to check if accuracies are significantly different between random classifier and poisson model.

### CODE REVIEW

**Comment by tkpratardan** [comment]
Points:
- The poisson model has no draw predictions.
- Very good hit rate for home_wins 
- Statistically significant accuracy difference (overall and for each class). How do we interpret the values for draw where model has no predictions at all?

File: research_amp/soccer_prediction/paired_t_test.py [hunk 1]
```diff
@@ -12,135 +12,484 @@
...
+                    df.columns, 
+                    f"The DataFrame does not contain the given columns: {actual_column}")
+    classes = ['home_win', 'draw', 'away_win']
+    hit_rates = {}
+    # Compute hit rates for each class.
+    for cls in classes:
+        hits = np.sum((df[actual_column] == cls) & (df[pred_column] == cls))
+        total = np.sum(df[actual_column] == cls)
+        hit_rate = hits / total if total > 0 else 0
+        ci = scps.norm.ppf(0.975) * np.sqrt((hit_rate * (1 - hit_rate)) / total) if total > 0 else 0
```
Comment (gpsaggese) [inline research_amp/soccer_prediction/paired_t_test.py:165]: let's pass the alpha here through the interface with a default value

File: research_amp/soccer_prediction/paired_t_test.py [hunk 2]
```diff
@@ -12,135 +12,484 @@
...
+def compute_hit_rate(
+                    df: pd.DataFrame(),
+                    pred_column: str,
+                    actual_column: str = "actual_outcome", 
+                    ) -> Dict:
+    """
+    Compute the hit rates for each class.
+    
+    :param df: df containing soccer match data and predictions.
+    :return: Hit rates dictionary.
```
Comment (gpsaggese) [inline research_amp/soccer_prediction/paired_t_test.py:149]: let's put an example of how the dict will look like

File: research_amp/soccer_prediction/paired_t_test.py [hunk 3]
```diff
@@ -12,135 +12,484 @@
...
+    overall_ci = scps.norm.ppf(0.975) * np.sqrt((overall_hit_rate * (1 - overall_hit_rate)) / overall_total)
+    hit_rates['overall'] = {
+        'hit_rate': overall_hit_rate,
+        'confidence_interval': (overall_hit_rate - overall_ci, overall_hit_rate + overall_ci)
+    }
+    # Return hit rates for each class and overall.
+    return hit_rates
+
+
+def compute_confusion_matrix_and_f1(
```
Comment (gpsaggese) [inline research_amp/soccer_prediction/paired_t_test.py:183]: linter

**Review by gpsaggese — COMMENTED** [review]
Very well done. Ofc still some nits from our flow, but the code is well done, uses libs well.

The code structure a bit wonky, but it's just a matter of working with us to learn how to partition the code.

Let's merge and do a PR to apply the changes.

I'll also find time to run it.

**Pull request merged**